feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.ts

```diff
@@ -0,0 +1,158 @@
+import { ConfiguratorAttributeBaseComponent } from './configurator-attribute-base.component';
+import { Directive, EventEmitter, Input, Output } from '@angular/core';
+import { Configurator } from '../../../../core/model/configurator.model';
+import { ConfigFormUpdateEvent } from '../../../form/configurator-form.event';
+import { ConfiguratorAttributeQuantityComponentOptions } from '../../quantity/configurator-attribute-quantity.component';
+import { map } from 'rxjs/operators';
+import { BehaviorSubject } from 'rxjs';
+import { ConfiguratorAttributeQuantityService } from '../../quantity/configurator-attribute-quantity.service';
+import { ConfiguratorPriceComponentOptions } from '../../../price/configurator-price.component';
+import { FormControl } from '@angular/forms';
+
+@Directive()
+// eslint-disable-next-line @angular-eslint/directive-class-suffix
+export abstract class ConfiguratorAttributeSingleSelectionBaseComponent extends ConfiguratorAttributeBaseComponent {
```

Comment:
Please add tests of this component, and clean tests of its children.

Reply:
Picked!

Reply:
Done!